scripts/cookie-auditor/run/index.js

```diff
@@ -0,0 +1,354 @@
+  try {
+    await fs.promises.access(path.join(__dirname, '../base-data/chromium.xlsx'))
+    // If it does, create a SheetJS workbook from it
+
+    baseCookieData = glob
+      .sync(path.join(__dirname, '../base-data/*.xlsx'))
+      .map((baseDataFile) => {
+        XLSX.readFileSync(
+          path.join(__dirname, `../base-data/${baseDataFile}.xlsx`)
+        )
+      })
+  } catch {
+    throw Error('No `base-data` file(s) found to compare against! Aborting.')
+  }
+
+  await createNewWorkbooks(pages)
+
+  // Add workbooks generated above to a new array, with browser type as label
+  const workbooks = glob
+    .sync(path.join(__dirname, '../out/cookie-data/*.xlsx'))
+    .map((path) => {
+      const browserTypeRegex = new RegExp(
+        /(?<=cookie-data\/)(.*)\w(?=.xlsx)/,
+        'gi'
+      )
+
+      return {
+        browserType: path.match(browserTypeRegex),
+        newCookieData: XLSX.readFileSync(path),
+      }
+    })
+
+  /**
+   * For each new workbook, compare its data to the base data
+   * and export the differences
+   */
+  for (const { browserType, newCookieData } of workbooks) {
+    await recordCookieDataComparison(baseCookieData, newCookieData, browserType)
+  }
+}
+
+run()
+
+// ==============
+// Functions used
+// ==============
+
+/**
+ * Creates Excel workbooks containing new cookie
+ * data gathered from headless Chromium and Firefox,
+ * using Playwright.
+ *
+ * Exports an `.xlsx` file per browser engine after
+ * successful gathering of cookie data.
+ *
+ * @param {Array} sites An array of objects containing
+ * a title and url for each of our sites
+ */
+async function createNewWorkbooks(sites) {
+  console.log(`Recording new cookie data into workbooks...`)
+
+  // TODO Add `webkit` to this browser type list once the TLS handshake error is solved
+  for (const browserType of [chromium, firefox]) {
+    const browserWb = XLSX.utils.book_new()
+
+    // Open a browser with the specified browser engine
+    console.log(`Launching ${browserType.name()} browser...`)
+
+    const browser = await browserType.launch()
+
+    // Collect data from each page and add it to the browser-specific workbook
+    console.log('Beginning cookie collection...')
+
+    for (const siteData of sites) {
+      await collectAndRecordData(siteData, browser, browserWb)
+    }
+
+    console.log(`Cookie collection complete for ${browserType.name()}!`)
+
+    // Cleanup: close the Playwright browser and write the workbook to file
+    console.log(`Closing ${browserType.name()} browser...`)
+
+    await browser.close()
+
+    console.log(`Exporting ${browserType.name()} cookie data...`)
+
+    // If /cookie-data isn't yet created in /out, create it
+    try {
+      await fs.promises.access(path.join(__dirname, '../out/cookie-data'))
+    } catch {
+      fs.mkdir(path.join(__dirname, '../out/cookie-data'), (err) => {
+        if (err) {
+          throw err
+        }
+      })
+    }
```

Comment:
I figured these checks wouldn't be needed using `glob` or `XLSX.writeFileSync` (I use the latter on line 116), but they seem to be.  Perhaps there's a better way to write these 😅